### DISCUSSION TITLE
3D volume construction algorithm

### DISCUSSION THREAD

**Discussion by Nenumarokas** [nroduit/Weasis#594] [Q&A]
Hello,

I was wondering how the 2D layers of DICOM are converted to a volume. I would assume it uses something like alphashape, but in 3D?
I tried to find the implementation to no avail.

Additionally, how are the images filtered to preserve tissue but remove the whatever-else around it? I have this example of a heart:

![image](https://github.com/user-attachments/assets/51a19b9f-3ce4-4df0-9b6f-6c23a3ca8fb8)
![image](https://github.com/user-attachments/assets/b232d232-09d4-40bf-bd12-90bc51d626a6)

**Comment by nroduit** [comment]
You should start by looking at the [documentation](https://weasis.org/en/tutorials/dicom-3d-viewer/). The implementation uses OpenGL and shaders.
